events/source/EventQueue.cpp

```diff
@@ -41,9 +41,19 @@ EventQueue::~EventQueue()
     equeue_destroy(&_equeue);
 }
 
-void EventQueue::dispatch(int ms)
+void EventQueue::dispatch(duration ms)
```

Comment:
This is a bit late, but the other possibility here is that maybe these could align better with the RTOS APIs if they were `dispatch_for(duration)` and then maybe just `dispatch()`?

Not sure which is better - just throwing it in.

Reply:
@evedon what are your though on Kevin's on the above?
@kjbracey-arm so would you then see us having:
dispatch_for(duration)
dispatch_forever()
and
dispatch()  ?

Reply:
I think `dispatch_for` is good for consistency with other relative-time-limited calls. It should be that.

After thinking a bit more, I think you should stay with `dispatch_forever`. I was thinking of other unsuffixed calls like `acquire` or `lock` that could wait indefinitely, but they do have a clear target at which point they should finish in the verb. "Dispatch" isn't like that, so makes sense to be specific about "forever".

Reply:
Yes, `dispatch_for(duration)` instead of `dispatch(duration)` better aligns with the RTOS APIs

Reply:
Note we will have to deprecate the dispatch() function then in order to add dispatch_for()